contextily/plotting.py

```diff
@@ -104,6 +105,13 @@ def add_basemap(
 
     """
     xmin, xmax, ymin, ymax = ax.axis()
+
+    if isinstance(source, str):
+        try:
+            source = providers.query_name(source)
```

Comment:
Quick question: where's `query_name` live here? I can't find it on the code base

Reply:
Nope, that is a method of `xyzservices.Bunch`. And since `providers` is a Bunch, it comes from there.